### PR TITLE
simple prius simulator example

### LINKED ISSUE #1578: move prius model into drake examples
- [x] port urdf w/ ackerman steering, suspension springs, and cylinder wheel collision geometry
- [x] port lcm keyboard/joystick interface app
- [x] port @crogers3 throttle/steering pd controller

### PR DESCRIPTION
**Pull request by RussTedrake**
resolves #1578 

- port urdf w/ ackerman steering, suspension springs, and cylinder wheel collision geometry from @Sproulx 
- port @crogers3 throttle/steering pd controller (and adds plumbing for PD controllers to the C++ systems)
- implements joint limits and rigid body springs
- adds a simple command line publisher for the steering command

lots of small supporting changes to the evolving c++ toolchain.

### CODE REVIEW

File: drake/examples/Cars/simulateLCM.cpp [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+    return *this;
+  }
+
+  friend Eigen::Vector3d toEigen(const DrivingCommand<ScalarType>& vec) {
+    Eigen::Vector3d x;
+    x << vec.steering_angle, vec.throttle, vec.brake;
+    return x;
+  }
+
+  friend std::string getCoordinateName(const DrivingCommand<ScalarType>& vec, unsigned int index) {
```
Comment (hongkai-dai) [inline drake/examples/Cars/simulateLCM.cpp]: Why we need to pass in `vec` here? It is not used in the function

File: drake/examples/Cars/simulateLCM.cpp [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+    return "error";
+  }
+  const static int RowsAtCompileTime = 3;
+
+  ScalarType steering_angle;
+  ScalarType throttle;
+  ScalarType brake;
+};
+
+bool decode(const drake::lcmt_driving_control_cmd_t& msg, double& t, DrivingCommand<double>& x) {
```
Comment (hongkai-dai) [inline drake/examples/Cars/simulateLCM.cpp]: so `decode` will always return true? How does it handle parsing error?

File: drake/systems/Simulation.h [hunk 3]
```diff
@@ -16,12 +16,14 @@ namespace Drake {
 
   // simulation options
   struct SimulationOptions {
-    double realtime_factor;  // 1 means try to run at realtime speed, < 0 is run as fast as possible
+    double realtime_factor;  // 1 means try to run at realtime speed, 0 is run as fast as possible, < 0 means use default
```
Comment (hongkai-dai) [inline drake/systems/Simulation.h]: Are we going to specify the integration method inside `SimulationOptions` in the future?

File: drake/systems/plants/RigidBodySystem.cpp [hunk 4]
```diff
@@ -32,31 +37,38 @@ RigidBodySystem::StateVector<double> RigidBodySystem::dynamics(const double& t,
...
+  { // apply joint limit forces
+    for (auto const& b : tree->bodies) {
+      if (!b->hasParent()) continue;
+      auto const& joint = b->getJoint();
+      if (joint.getNumPositions()==1 && joint.getNumVelocities()==1) {  // taking advantage of only single-axis joints having joint limits makes things easier/faster here
+        double qmin = joint.getJointLimitMin()(0), qmax = joint.getJointLimitMax()(0);
+        // tau = k*(qlimit-q) - b(qdot)
+        if (q(b->position_num_start)<qmin)
+          C(b->velocity_num_start) -= penetration_stiffness*(qmin-q(b->position_num_start)) - penetration_damping*v(b->velocity_num_start);
+        if (q(b->position_num_start)>qmax)
```
Comment (hongkai-dai) [inline drake/systems/plants/RigidBodySystem.cpp]: `else if` here? Out of curiosity, if `qmin` is very close to `qmax`, how shall we guarantee that the joint limit force does not cause the joint to oscillate? Can we verify a certain choice of stiffness and damping should always work for some joint limit and time step?

**Comment by hongkai-dai** [comment]
code looks awesome!

File: drake/systems/Simulation.h (hunk 3 above)
Comment (RussTedrake) [inline drake/systems/Simulation.h]: absolutely.  and I think that future is coming fast.  :)

File: drake/examples/Cars/simulateLCM.cpp (hunk 1 above)
Comment (RussTedrake) [inline drake/examples/Cars/simulateLCM.cpp]: it's required to model the vector concept.  
```
  /** getCoordinateName()
   * @brief Implements default coordinate names for a generic vector.  Overload this to name your coordinates.
   * @concept{vector_concept}
   */
```

File: drake/examples/Cars/simulateLCM.cpp (hunk 2 above)
Comment (RussTedrake) [inline drake/examples/Cars/simulateLCM.cpp]: same as above.  other decode methods might fail, this one is so simple that failure is not considered an option.

File: drake/systems/plants/RigidBodySystem.cpp (hunk 4 above)
Comment (RussTedrake) [inline drake/systems/plants/RigidBodySystem.cpp]: done

**Comment by RussTedrake** [comment]
thanks @hongkai-dai

**Pull request merged**